### PR TITLE
Add support for eslint flat config

### PR DESCRIPTION
**Pull request by OlivierZal**

### CODE REVIEW

**Comment by OlivierZal** [comment]
@ljharb, here is the PR to make https://github.com/airbnb/javascript/pull/2818 work.

File: config/flat.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const * as importPlugin = require('../src/index');
```
Comment (ljharb) [inline config/flat.js:1]:
```suggestion
const importPlugin = require('../src/index');
```

**Comment by ljharb** [comment]
this may also need to be coordinated with #2829.

**Comment by OlivierZal** [comment]
@ljharb, am I suppose to solve the failing tests?

**Comment by ljharb** [comment]
@bmish this PR adds a new entry point; what needs to be done for the docs generator to recognize it?

**Comment by OlivierZal** [comment]
Maybe because there is no `rules` in this entrypoint?

**Comment by OlivierZal** [comment]
https://github.com/bmish/eslint-doc-generator/blob/d27c509ef0dda1cdf5dc911d9eb6bec57556e60a/lib/package-json.ts#L64

**Comment by OlivierZal** [comment]
@ljharb, I think it's because the plugin value is not string: https://github.com/bmish/eslint-doc-generator/blob/d27c509ef0dda1cdf5dc911d9eb6bec57556e60a/lib/package-json.ts#L57

**Comment by OlivierZal** [comment]
https://github.com/bmish/eslint-doc-generator/blob/d27c509ef0dda1cdf5dc911d9eb6bec57556e60a/lib/package-json.ts#L64

**Comment by ljharb** [comment]
hm, maybe it's not yet compatible with flat config?

**Comment by OlivierZal** [comment]
That's my guess 😢

**Comment by ljharb** [comment]
um, I’m confused, why did you delete your forks, making these PRs unrecoverable??

**Comment by bmish** [comment]
I have an issue related to this: https://github.com/bmish/eslint-doc-generator/issues/376. Any thoughts on what needs to be done to support flat config?

**Comment by ljharb** [comment]
Seems like it relies on the plugin name in an entry point being a string.

**Pull request closed without merging**